src/playdate/sound.nim

```diff
@@ -221,25 +249,28 @@ proc rate*(this: SamplePlayer): float32 =
     privateAccess(PlaydateSoundSampleplayer)
     return playdate.sound.sampleplayer.getRate(this.resource).float32
 
-proc privateSampleFinishCallback(soundSourcePtr: SoundSourcePtr, userData: pointer) {.cdecl, raises: [].} =
+proc privateSamplePlayerFinishCallback(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl, raises: [].} =
     try:
-        let samplePlayer = cast[SamplePlayer](userdata)[]
-        if samplePlayer.callback != nil:
-            samplePlayer.callback(userData)
+        let samplePlayer = cast[SamplePlayer](userdata)
+        if samplePlayer.finishCallback != nil:
```

Comment:
yeah good one on specifying the name. There will be a loopCallback too, after all.